gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -180,6 +182,16 @@ def self.note_module_deals_with_final(mod)
     @modules_with_final[mod.singleton_class]
   end
 
+  def self._with_declaring_final_method(&blk)
```

Comment:
Can we pick a name that makes it a little more clear that this is a private API? like `_INTERNAL` or something? Can we also add a comment describing expected usage?